### DISCUSSION TITLE
Guild Ranks/Priorities

### DISCUSSION THREAD

**Discussion by BlueSnapple** [HypixelDev/PublicAPI#463] [Q&A] [answered]
I'm trying to make a plugin using the api and part of it hooks into a permissions plugin and creates groups based off of a guild's rank. I'm unable to locate methods in the api to create a solid method of automating this sort of ranking system. Is there any way I can get part of the api accessing the priority method for guild ranks and I can sort it from there? or maybe just an easy way to grab all ranks in a guild and have it listed in order of priority or numbered? I've looked at the json created from getPlayer data and getGuild data and even getResource(GuildPermissions). It seems like getResource(GuildPermissions) is still being worked on and will implement the things I'd like in the future? I'm not sure. Anything helps, thank you!

**Comment by TheNullicorn** [comment]
In the [guild endpoint](https://api.hypixel.net/#tag/Player-Data/paths/~1guild/get) under `guild.ranks`, each element should have a `priority` key indicating the rank's order in the hierarchy. It's not perfectly sequential—I believe there can be gaps if a rank was deleted—but the order should be what you're looking for (lower number = lower priority, and vice-versa). Specific rank permissions don't appear to be in the API anymore, so priority may be as close as you can get for now.

<details>
<summary>An aside...</summary>
Your API tutorials from a few years back were what got me into using the API in the first place - really appreciate those!
</details>

**Reply by BlueSnapple** [reply]
So, you are meaning --> getGuild(name/uuid) and grabbing the guild reply and inside the json it would list a "guild.ranks" option and contain priorities for each rank? Sorry if that was a rough explanation. If what I just tried explaining, is what you are meaning, then I don't think that functionality is in the java portion of the api? I'll show you an example of my guild reply

`Guild{_id='5d75253e77ce84059e58eff3', name='Instant Karma', description='null', tag='KARMA', tagColor='DARK_AQUA', publiclyListed=true, banner=null, members=[Member{uuid=c1b08711-3ddc-4a39-9e10-48ae7f5a9eeb, rank=Guild Master, joined=2019-09-08T11:58:54.892-04:00[America/New_York]}, Member{uuid=53f6f387-2dd3-446b-9f3e-899ab7f0c4b8, rank=WOAH, joined=2019-09-28T13:18:46.121-04:00[America/New_York]}, Member{uuid=0e12bd11-221d-4f42-8598-fd910fe8c5de, rank=Rookie, joined=2019-10-21T17:15:23.997-04:00[America/New_York]}], coins=0, coinsEver=0, created=2019-09-08T11:58:54.892-04:00[America/New_York], joinable=null, exp=36696333, legacyRanking=0}`

It does indeed show a member's rank like I'd expect it to, but there's no showing of a priority. Can you point me in the right direction? I've checked `api.getResource(ResourceType.GUILDS_PERMISSION).get()` and it doesn't seem to have any useful information. Here's what I get by looking it up 

`{"success":true,"lastUpdated":1570754198669,"permissions":[{"en_us":{"name":"Modify Guild Name","description":"Change the guild's name.","item":{"name":"name_tag"}}},{"en_us":{"name":"Modify Guild MOTD","description":"Change the guild's message of the day.","item":{"name":"paper"}}},{"en_us":{"name":"Modify Guild Tag","description":"Change the guild's tag.","item":{"name":"sign"}}},{"en_us":{"name":"Change ranks","description":"Promote or demote members (up to their own rank).","item":{"name":"crafting_table"}}},{"en_us":{"name":"Guild Finder options","description":"Change how the guild is shown in the Guild Finder, if at all.","item":{"name":"book"}}},{"en_us":{"name":"Officer Chat","description":"Send and receive messages in the officer chat.","item":{"name":"emerald"}}},{"en_us":{"name":"Kick Members","description":"Kick members from the guild.","item":{"name":"barrier"}}},{"en_us":{"name":"Mute Members","description":"Mute guild members.","item":{"name":"redstone"}}},{"en_us":{"name":"Invite members","description":"Invite members to the guild.","item":{"name":"arrow"}}},{"en_us":{"name":"Audit Log","description":"View the audit log.","item":{"name":"lever"}}},{"en_us":{"name":"View Stats","description":"View a guild member's stats.","item":{"name":"diamond"}}},{"en_us":{"name":"Guild Party","description":"Start a guild party.","item":{"name":"chest"}}},{"en_us":{"name":"Guild Settings","description":"Change the guild settings.","item":{"name":"comparator"}}},{"en_us":{"name":"Change Guild Discord","description":"Change the guild's Discord join link.","item":{"name":"filled_map"}}}]}`

All I really need in my particular application is the rank priority, since I'd be able to sort the guild ranks accordingly. Guild master is always Guild Master and so I'd just need the other 4 "dynamic variables" I guess you might call them. Anyways, thanks for the response! Let me know if you can add onto any thing else.

<details>
  <summary>An aside...</summary>
  Hey, thanks! I'm glad the tutorials were of use to you :D I'm actually getting back into coding again and I'll be making an update/more of them, soon.
</details>

**Reply by TheNullicorn** [reply]
It never occurred to me that the `ranks` field was left out of the Java API. It's been in the REST one for a while now. If you'd like I could PR it when I have the chance, because that seems like a pretty big oversight.

e.g.
![rank list](https://user-images.githubusercontent.com/64789500/132263258-69e99a1c-8590-4a0c-9bb0-60fcc0c5e474.png)
(notice the `priority` field)

**Reply by BlueSnapple** [reply]
Yeah, that would be sweet! I thought maybe I was just missing something.

**Comment by BlueSnapple** [comment, marked as answer]
The Hypixel Java API has not implemented the method to grab guild rank priorities. Pending - pull request

**Reply by TheNullicorn** [reply]
Just opened #466 which should do what you need, plus a little extra hopefully! :)

**Reply by BlueSnapple** [reply]
You're the bomb! 💣 Much appreciated 😎

**Reply by TheNullicorn** [reply]
If you're still following this, it just got merged in [4.1](https://github.com/HypixelDev/PublicAPI/releases/tag/4.1). Good luck with your plugin! :)

**Reply by BlueSnapple** [reply]
Oh, sweet! I ended up using part of the REST in Java as a work around. Thanks for that, man. Plug-in is all done 😁

**Reply by BlueSnapple** [reply]
Now, I can convert it back to Java 😎
